pkg/controller/core/workload_controller.go

```diff
@@ -470,6 +471,18 @@ func (r *WorkloadReconciler) triggerDeactivationOrBackoffRequeue(ctx context.Con
 	for backoff.Steps > 0 {
 		waitDuration = backoff.Step()
 	}
+
+	maxWaitDuration := time.Duration(r.waitForPodsReady.requeuingBackoffMaxSeconds) * time.Second
+	if waitDuration > maxWaitDuration {
+		wl.Spec.Active = ptr.To(false)
+		if err := r.client.Update(ctx, wl); err != nil {
+			return false, err
+		}
+		r.recorder.Eventf(wl, corev1.EventTypeNormal, kueue.WorkloadEvictedByDeactivation,
+			"Deactivated Workload %q by reached re-queue backoffMaxSeconds", klog.KObj(wl))
+		return true, nil
+	}
```

Comment:
I think the desired behaviour is not to have wait times more then `BackoffMaxSeconds`

```suggestion
	maxWaitDuration := time.Duration(r.waitForPodsReady.requeuingBackoffMaxSeconds) * time.Second
	for backoff.Steps > 0 {
		waitDuration = backoff.Step()
		if waitDuration > maxWaitDuration {
			waitDuration > maxWaitDuration
			break
		}
	}
```

Reply:
Got it. Thanks